<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
 <HEAD>
   <TITLE> [mod_python] Compile problems - Mod_Python 3.1.4
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:mod_python%40modpython.org?Subject=%5Bmod_python%5D%20Compile%20problems%20-%20Mod_Python%203.1.4&In-Reply-To=427FC0A9.2050007%40verizon.net">
   <META NAME="robots" CONTENT="index,nofollow">
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="018022.html">
   <LINK REL="Next"  HREF="018026.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[mod_python] Compile problems - Mod_Python 3.1.4</H1>
<table border=0 width="100%"><tr><td valign="top">
    <B>Jim Gallacher</B> 
    <A HREF="mailto:mod_python%40modpython.org?Subject=%5Bmod_python%5D%20Compile%20problems%20-%20Mod_Python%203.1.4&In-Reply-To=427FC0A9.2050007%40verizon.net"
       TITLE="[mod_python] Compile problems - Mod_Python 3.1.4">jg.lists at sympatico.ca
       </A><BR>
    <I>Mon May  9 17:22:20 EDT 2005</I>
    <P><UL>
        <LI>Previous message: <A HREF="018022.html">[mod_python] Compile problems - Mod_Python 3.1.4
</A></li>
        <LI>Next message: <A HREF="018026.html">[mod_python] Compile problems - Mod_Python 3.1.4
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#18024">[ date ]</a>
              <a href="thread.html#18024">[ thread ]</a>
              <a href="subject.html#18024">[ subject ]</a>
              <a href="author.html#18024">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>Frank,

Frank Gruman wrote:
&gt;<i> Nicolas,
</I>&gt;<i> 
</I>&gt;<i> The script you are running is automatically included as part of the make 
</I>&gt;<i> process on Linux.
</I>&gt;<i> 
</I>&gt;<i> But I am willing to try anything...
</I>
OK, here's something. :)

&gt;<i> Running the script with the method you specify returns the same ending 
</I>&gt;<i> error code that I found earlier (about the APXS definition not found).  
</I>&gt;<i> Thanks for the idea, though.
</I>&gt;<i> 
</I>&gt;<i> Anyone have any idea why the setup.py script is not able to pass the 
</I>&gt;<i> APXS value from config.status back to setup.py (getconfigure_option 
</I>&gt;<i> function).  I have verified that there is a value assigned in 
</I>&gt;<i> config.status (/usr/local/apache2/bin/apxs).
</I>
I think there is a problem with the regular expression which searches 
for APXS in config.status. (I'm getting the same error on Debian as you 
are on SLES).

Setup.py is using the following re:
r = re.compile('s, @%s@, (?P&lt;OPTION_STRING&gt;[^,]+), ' % (option_name))

While my config.status output looks like this:
<A HREF="http://mailman.modpython.org/mailman/listinfo/mod_python">s, at APXS</A>@,/usr/bin/apxs2,;t t

Notice the lack of spaces after the commas? I wonder if the output of 
configure is somehow platform dependant, which is why Nicolas has not 
seen a problem?

I've been trying to tweak the re to make it work, but so far it's mainly 
been an exercise in banging head against the monitor. (Somebody really 
should come up with a padded monitor - they'd make a fortune). I both 
love and hate regular expressions at the same time.

Anyway, as a quick fix you could apply the following patch to 
setup.py.in, and do the configure + make cycle again.

Index: setup.py.in
===================================================================
--- setup.py.in	(revision 169349)
+++ setup.py.in	(working copy)
@@ -62,7 +62,8 @@

  def getapxs_location():
      &quot;&quot;&quot;finds the location of apxs from the config.status file&quot;&quot;&quot;
-    return getconfigure_option(&quot;APXS&quot;)
+    #return getconfigure_option(&quot;APXS&quot;)
+    return '/usr/local/apache2/bin/apxs'

  def getapxs_option(option):
      APXS = getapxs_location()


Also, I don't know if you resolved the psp_flex issue, but the 
src/Makefile.in hard codes LEX=/usr/local/bin/flex, which fails on my 
system since flex is located in /usr/bin. A quick edit of the Makefile 
took care of that for me.

If your luck is anything like mine the next thing you'll have to deal 
with is the linker not finding libapr.

Personally, I've avoided these issues by using the setup.py included 
with Debian, even when compiling from the svn trunk.

Regards,
Jim


&gt;<i> Oh - also realized I didn't provide OS - SLES 9.
</I>&gt;<i> 
</I>&gt;<i> Regards,
</I>&gt;<i> Frank
</I>&gt;<i> 
</I>&gt;<i> Nicolas Lehuen wrote:
</I>&gt;<i> 
</I>&gt;&gt;<i>Hi,
</I>&gt;&gt;<i>
</I>&gt;&gt;<i>Just a wild guess, as I'm building under Win32... Did you try :
</I>&gt;&gt;<i>
</I>&gt;&gt;<i>python setup.py install
</I>&gt;&gt;<i>
</I>&gt;&gt;<i>?
</I>&gt;&gt;<i>
</I>&gt;&gt;<i>This works under Win32.
</I>&gt;&gt;<i>
</I>&gt;&gt;<i>Regards,
</I>&gt;&gt;<i>
</I>&gt;&gt;<i>Nicolas
</I>&gt;&gt;<i>
</I>&gt;&gt;<i>2005/5/9, Frank Gruman &lt;<A HREF="http://mailman.modpython.org/mailman/listinfo/mod_python">fgatwork at verizon.net</A>&gt;:
</I>&gt;&gt;<i>  
</I>&gt;&gt;<i>
</I>&gt;&gt;&gt;<i>Hello all,
</I>&gt;&gt;&gt;<i>
</I>&gt;&gt;&gt;<i>I am trying to test the latest Apache dev version, and found an issue when trying to compile mod_python 3.1.4 with it.
</I>&gt;&gt;&gt;<i>
</I>&gt;&gt;&gt;<i>I originally tired using Apache 2.1.5-dev (from tip on Friday, 06-May) with mod_python 3.1.4 and Python 2.4.1.  When this failed, I thought that there might have been a resolution in the current mod_python tip, so pulled the mod_python tip as of 09-May @ 13:00.  Still ran into the same errors.
</I>&gt;&gt;&gt;<i>
</I>&gt;&gt;&gt;<i>The system goes through the ./configure without any issues.  Please see the steps / output below.  It gets quite long, so please bear with me.
</I>&gt;&gt;&gt;<i>
</I>&gt;&gt;&gt;<i>*  Configure *****************************************
</I>&gt;&gt;&gt;<i>toybox:/sources/mod_python-3.1.4 # ./configure --with-apxs=/usr/local/apache2/bin/apxs
</I>&gt;&gt;&gt;<i>checking for gcc... gcc
</I>&gt;&gt;&gt;<i>checking for C compiler default output... a.out
</I>&gt;&gt;&gt;<i>checking whether the C compiler works... yes
</I>&gt;&gt;&gt;<i>checking whether we are cross compiling... no
</I>&gt;&gt;&gt;<i>checking for suffix of executables...
</I>&gt;&gt;&gt;<i>checking for suffix of object files... o
</I>&gt;&gt;&gt;<i>checking whether we are using the GNU C compiler... yes
</I>&gt;&gt;&gt;<i>checking whether gcc accepts -g... yes
</I>&gt;&gt;&gt;<i>checking for ar... ar
</I>&gt;&gt;&gt;<i>checking for a BSD-compatible install... /usr/bin/install -c
</I>&gt;&gt;&gt;<i>checking whether make sets ${MAKE}... yes
</I>&gt;&gt;&gt;<i>checking for main in -lm... yes
</I>&gt;&gt;&gt;<i>checking for gcc option to accept ANSI C... none needed
</I>&gt;&gt;&gt;<i>checking for an ANSI C-conforming const... yes
</I>&gt;&gt;&gt;<i>checking your blood pressure... a bit high, but we can proceed
</I>&gt;&gt;&gt;<i>configure: checking whether apxs is available...
</I>&gt;&gt;&gt;<i>checking for --with-apxs... /usr/local/apache2/bin/apxs executable, good
</I>&gt;&gt;&gt;<i>checking Apache version... 2.1.5-dev
</I>&gt;&gt;&gt;<i>checking for Apache libexec directory... /usr/local/apache2/modules
</I>&gt;&gt;&gt;<i>checking for Apache include directory... -I/usr/local/apache2/include
</I>&gt;&gt;&gt;<i>checking for --with-python... no
</I>&gt;&gt;&gt;<i>checking for python... /usr/bin/python
</I>&gt;&gt;&gt;<i>checking Python version... 2.4
</I>&gt;&gt;&gt;<i>checking Python install prefix... /usr
</I>&gt;&gt;&gt;<i>checking for Py_NewInterpreter in -lpython2.4... yes
</I>&gt;&gt;&gt;<i>checking what libraries Python was linked with... -lpython2.4   -lpthread -ldl  -lutil   -lm
</I>&gt;&gt;&gt;<i>checking linker flags used to link Python...
</I>&gt;&gt;&gt;<i>checking where Python include files are... -I/usr/include/python2.4
</I>&gt;&gt;&gt;<i>configure: creating ./config.status
</I>&gt;&gt;&gt;<i>config.status: creating Makefile
</I>&gt;&gt;&gt;<i>config.status: creating src/Makefile
</I>&gt;&gt;&gt;<i>config.status: creating Doc/Makefile
</I>&gt;&gt;&gt;<i>config.status: creating test/testconf.py
</I>&gt;&gt;&gt;<i>config.status: creating dist/setup.py
</I>&gt;&gt;&gt;<i>config.status: creating dist/Makefile
</I>&gt;&gt;&gt;<i>toybox:/sources/mod_python-3.1.4 #
</I>&gt;&gt;&gt;<i>****************************************************
</I>&gt;&gt;&gt;<i>
</I>&gt;&gt;&gt;<i>No warnings or errors.  Looks good.  Let's go on...
</I>&gt;&gt;&gt;<i>
</I>&gt;&gt;&gt;<i>* 1st Make *****************************************
</I>&gt;&gt;&gt;<i>toybox:/sources/mod_python-3.1.4 # make
</I>&gt;&gt;&gt;<i>make[1]: Entering directory `/sources/mod_python-3.1.4/src'
</I>&gt;&gt;&gt;<i>
</I>&gt;&gt;&gt;<i>Compiling for DSO.
</I>&gt;&gt;&gt;<i>
</I>&gt;&gt;&gt;<i>    
</I>&gt;&gt;&gt;<i>
</I>&gt;<i> &lt;snipped log as it was getting to big for the message board&gt;
</I>&gt;<i> 
</I>&gt;&gt;&gt;<i>/usr/local/apache2/build/libtool --silent --mode=link gcc -o mod_python.la  -rpath /usr/local/apache2/modules -module -avoid-version    hlistobject.lo hlist.lo filterobject.lo connobject.lo serverobject.lo util.lo tableobject.lo requestobject.lo _apachemodule.lo mod_python.lo -L/usr/lib/python2.4/config -Xlinker -export-dynamic -lm -lpython2.4 -lpthread -ldl -lutil -lm
</I>&gt;&gt;&gt;<i>make[1]: Leaving directory `/sources/mod_python-3.1.4/src'
</I>&gt;&gt;&gt;<i>make[1]: Entering directory `/sources/mod_python-3.1.4/dist'
</I>&gt;&gt;&gt;<i>ln -s ../lib/python/mod_python
</I>&gt;&gt;&gt;<i>ln -s ../src
</I>&gt;&gt;&gt;<i>make[2]: Entering directory `/sources/mod_python-3.1.4/src'
</I>&gt;&gt;&gt;<i>/usr/local/bin/flex -R -opsp_parser.c --header-file=include/psp_flex.h psp_parser.l
</I>&gt;&gt;&gt;<i>make[2]: /usr/local/bin/flex: Command not found
</I>&gt;&gt;&gt;<i>make[2]: *** [psp_parser.c] Error 127
</I>&gt;&gt;&gt;<i>make[2]: Leaving directory `/sources/mod_python-3.1.4/src'
</I>&gt;&gt;&gt;<i>make[1]: *** [build] Error 2
</I>&gt;&gt;&gt;<i>make[1]: Leaving directory `/sources/mod_python-3.1.4/dist'
</I>&gt;&gt;&gt;<i>make: *** [do_dso] Error 2
</I>&gt;&gt;&gt;<i>toybox:/sources/mod_python-3.1.4 #
</I>&gt;&gt;&gt;<i>***************************************************
</I>&gt;&gt;&gt;<i>
</I>&gt;&gt;&gt;<i>It appears I the system could not find psp_parser.c.  This is not in the src directory.  I looked at the SVN repository, and found it there.  Not sure why it didn't come with the initial checkout, but I manually downloaded it.  Let's try to do a MAKE again.
</I>&gt;&gt;&gt;<i>
</I>&gt;&gt;&gt;<i>* 2nd Make ****************************************
</I>&gt;&gt;&gt;<i>
</I>&gt;&gt;&gt;<i>toybox:/sources/mod_python-3.1.4 # make clean &amp;&amp; make
</I>&gt;&gt;&gt;<i>
</I>&gt;&gt;&gt;<i>    
</I>&gt;&gt;&gt;<i>
</I>&gt;<i> &lt;snipped log as it was getting to big for the message board&gt;
</I>&gt;<i> 
</I>&gt;<i> 
</I>&gt;&gt;&gt;<i>/usr/local/apache2/build/libtool --silent --mode=link gcc -o mod_python.la  -rpath /usr/local/apache2/modules -module -avoid-version    hlistobject.lo hlist.lo filterobject.lo connobject.lo serverobject.lo util.lo tableobject.lo requestobject.lo _apachemodule.lo mod_python.lo -L/usr/lib/python2.4/config -Xlinker -export-dynamic -lm -lpython2.4 -lpthread -ldl -lutil -lm
</I>&gt;&gt;&gt;<i>make[1]: Leaving directory `/sources/mod_python-3.1.4/src'
</I>&gt;&gt;&gt;<i>make[1]: Entering directory `/sources/mod_python-3.1.4/dist'
</I>&gt;&gt;&gt;<i>ln -s ../lib/python/mod_python
</I>&gt;&gt;&gt;<i>make[2]: Entering directory `/sources/mod_python-3.1.4/src'
</I>&gt;&gt;&gt;<i>make[2]: `psp_parser.c' is up to date.
</I>&gt;&gt;&gt;<i>make[2]: Leaving directory `/sources/mod_python-3.1.4/src'
</I>&gt;&gt;&gt;<i>/usr/bin/python setup.py build
</I>&gt;&gt;&gt;<i>Traceback (most recent call last):
</I>&gt;&gt;&gt;<i>  File &quot;setup.py&quot;, line 138, in ?
</I>&gt;&gt;&gt;<i>    ModPyModule = ModPyExtension(getmp_srcdir(), [getmp_includedir(), getapache_includedir()], [getapache_libdir()])
</I>&gt;&gt;&gt;<i>  File &quot;setup.py&quot;, line 80, in getapache_includedir
</I>&gt;&gt;&gt;<i>    return getapxs_option(&quot;INCLUDEDIR&quot;)
</I>&gt;&gt;&gt;<i>  File &quot;setup.py&quot;, line 68, in getapxs_option
</I>&gt;&gt;&gt;<i>    APXS = getapxs_location()
</I>&gt;&gt;&gt;<i>  File &quot;setup.py&quot;, line 65, in getapxs_location
</I>&gt;&gt;&gt;<i>    return getconfigure_option(&quot;APXS&quot;)
</I>&gt;&gt;&gt;<i>  File &quot;setup.py&quot;, line 47, in getconfigure_option
</I>&gt;&gt;&gt;<i>    raise AssertionError(&quot;unable to find @%s@ definition in %s&quot;, (option_name, config_status_file))
</I>&gt;&gt;&gt;<i>AssertionError: ('unable to find @%s@ definition in %s', ('APXS', '/sources/mod_python-3.1.4/config.status'))
</I>&gt;&gt;&gt;<i>make[1]: *** [build] Error 1
</I>&gt;&gt;&gt;<i>make[1]: Leaving directory `/sources/mod_python-3.1.4/dist'
</I>&gt;&gt;&gt;<i>make: *** [do_dso] Error 2
</I>&gt;&gt;&gt;<i>toybox:/sources/mod_python-3.1.4 #
</I>&gt;&gt;&gt;<i>****************************************************
</I>&gt;&gt;&gt;<i>
</I>&gt;&gt;&gt;<i>So - now it seems to have problems reading the value I passed for the APXS parameter.  And I can't figure out why.
</I>&gt;&gt;&gt;<i>
</I>&gt;&gt;&gt;<i>Any ideas??
</I>&gt;&gt;&gt;<i>
</I>&gt;&gt;&gt;<i>Regards,
</I>&gt;&gt;&gt;<i>Frank
</I>&gt;&gt;&gt;<i>_______________________________________________
</I>&gt;&gt;&gt;<i>Mod_python mailing list
</I>&gt;&gt;&gt;<i><A HREF="http://mailman.modpython.org/mailman/listinfo/mod_python">Mod_python at modpython.org</A>
</I>&gt;&gt;&gt;<i><A HREF="http://mailman.modpython.org/mailman/listinfo/mod_python">http://mailman.modpython.org/mailman/listinfo/mod_python</A>
</I>&gt;&gt;&gt;<i>
</I>&gt;&gt;&gt;<i>    
</I>&gt;&gt;&gt;<i>
</I>&gt;&gt;<i>
</I>&gt;&gt;<i>  
</I>&gt;&gt;<i>
</I>&gt;<i> 
</I>&gt;<i> ------------------------------------------------------------------------
</I>&gt;<i> 
</I>&gt;<i> _______________________________________________
</I>&gt;<i> Mod_python mailing list
</I>&gt;<i> <A HREF="http://mailman.modpython.org/mailman/listinfo/mod_python">Mod_python at modpython.org</A>
</I>&gt;<i> <A HREF="http://mailman.modpython.org/mailman/listinfo/mod_python">http://mailman.modpython.org/mailman/listinfo/mod_python</A>
</I>
</PRE>





<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="018022.html">[mod_python] Compile problems - Mod_Python 3.1.4
</A></li>
	<LI>Next message: <A HREF="018026.html">[mod_python] Compile problems - Mod_Python 3.1.4
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#18024">[ date ]</a>
              <a href="thread.html#18024">[ thread ]</a>
              <a href="subject.html#18024">[ subject ]</a>
              <a href="author.html#18024">[ author ]</a>
         </LI>
       </UL>

</td>
<td align="right" valign="top">
<script type="text/javascript"><!--
google_ad_client = "pub-9718360309690383";
google_ad_width = 120;
google_ad_height = 600;
google_ad_format = "120x600_as";
google_color_border = "CCCCCC";
google_color_bg = "FFFFFF";
google_color_link = "000000";
google_color_url = "666666";
google_color_text = "333333";
//--></script>
<script type="text/javascript"
  src="DISABLEhttp://pagead2.googlesyndication.com/pagead/show_ads.js">
</script>
</td>
</table>

<hr>
<a href="http://mailman.modpython.org/mailman/listinfo/mod_python">More information about the Mod_python
mailing list</a><br>
</body></html>
